src/transport_ofi.c

```diff
@@ -1260,6 +1246,12 @@ int query_for_fabric(struct fabric_info *info)
     shmem_transport_ofi_mr_rma_event = (info->p_info->domain_attr->mr_mode & FI_MR_RMA_EVENT) != 0;
 #endif
 
+    if (shmem_internal_params.OFI_STX_NODE_MAX == -1) {
+        shmem_transport_ofi_tx_ctx_cnt = info->fabrics->domain_attr->tx_ctx_cnt;
+    } else {
+        shmem_transport_ofi_tx_ctx_cnt = shmem_internal_params.OFI_STX_NODE_MAX;
+    }
```

Comment:
Can we move this down to transport_init to avoid carrying the tx_ctx_cnt state across functions?

Reply:
I don't quite understand.  I believe this is inside transport_init right now (`query_for_fabric`).  Do you mean move it to `transport_startup`?  If so, I think that would be fine.

Reply:
Yes, that's what I meant.  :)  Move resolving this parameter down to where it's consumed, so we don't need to carry the extra state between functions.